rvps/src/cache/mod.rs

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) 2022 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+//! Cache is responsible for storing verified Reference Values
+
+use crate::reference_value::ReferenceValue;
+
+use anyhow::Result;
+
+/// Interface of an Cache.
+/// We only provide a simple instance here which implements
+/// Cache. In more scenerios, RV should be stored in persistent
+/// storage, like database, file and so on. All of the mentioned
+/// forms will have the same interface as following.
+pub trait Cache {
+    /// Store a reference value
+    fn set(&mut self, name: String, rv: ReferenceValue) -> Result<()>;
```

Comment:
Would this return `Err` if the value already existed in the cache? If not, shouldn't this return the existing value? That might be useful to the caller, or not (they could discard it if not). Plus, returning the original value would help with uniting testing the code.